internal/controller/object/object.go

```diff
@@ -382,6 +388,16 @@ func getLastApplied(obj *v1alpha2.Object, observed *unstructured.Unstructured) (
 
 func (c *external) setObserved(obj *v1alpha2.Object, observed *unstructured.Unstructured) error {
 	var err error
+
+	if c.sanitizeSecrets {
+		if observed.GetKind() == "Secret" && observed.GetAPIVersion() == "v1" {
+			observed, err = sanitizeUnstructuredSecret(observed)
```

Comment:
We can leverage the [fieldpath](https://github.com/crossplane/crossplane-runtime/tree/master/pkg/fieldpath) library of crossplane-runtime instead, example: https://github.com/crossplane/function-sdk-go/blob/f466e52426a9316b99677a4d6409995e985b2c37/resource/composite/composite.go#L336

Reply:
updated to use fieldpath library.